virtcontainers/container.go

```diff
@@ -472,13 +472,29 @@ func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (
 		// These mounts are created in the shared dir
 		filename := fmt.Sprintf("%s-%s-%s", c.id, hex.EncodeToString(randBytes), filepath.Base(m.Destination))
 		mountDest := filepath.Join(hostSharedDir, c.sandbox.id, filename)
+		sourcePath := filepath.Join(guestSharedDir, filename)
 
-		if err := bindMount(m.Source, mountDest, false); err != nil {
-			return nil, err
+		epheVol := false
+		for _, devInfo := range c.config.DeviceInfos {
+			if devInfo.DevType == "e" && devInfo.HostPath == m.Source {
```

Comment:
Please define the new device type in device/config/config.go, and compare `m.Source` with `devInfo.ContainerPath` instead.

And for epheVol, you don't need to put it in the shared mounts. Let me explain how it works:

1. on kata cli side, add a device to container config, where devinfo.type is epheVol, and a Mount to container config, where `m.Source == devinfo.ContainerPath`
2. in kata runtime, skip epheVol by **NOT** adding it to `sharedMounts`
3. in kata_agent.go, create tmpfs storage where its mountpoint is devinfo.ContainerPath

This would follow the same storage/device devinfo semantics as other types of devices/storages.